tornado/tcpclient.py

```diff
@@ -93,7 +94,13 @@ def try_connect(self, addrs):
                 self.future.set_exception(self.last_error or
                                           IOError("connection failed"))
             return
-        future = self.connect(af, addr)
+        try:
+            future = self.connect(af, addr)
+        except socket.error as e:
```

Comment:
But it does need to do some of the other stuff that's in `on_connect_done`'s `except` block: it at least needs to set `self.last_error`, and maybe reset the timeout (but probably not). 

It might be better if `_create_stream` trapped any error raised by `socket.socket()` and wrapped it in a `Future`.